### PR TITLE
#625: Make Dataproc init script timeout configurable

### PR DESCRIPTION
**Pull request by MatthewBemis**
will go in after https://github.com/broadinstitute/firecloud-develop/pull/1393 and tests will fail until then

Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [x] I've documented my API changes in Swagger

In all cases:

- [ ] Get two thumbsworth of review and PO signoff if necessary
- [x] Verify all tests go green
- [ ] Squash and merge; you can delete your branch after this
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

**Review by vkumra-broad — APPROVED** [review]

File: src/main/resources/reference.conf [hunk 1]
```diff
@@ -3,6 +3,7 @@
 dataproc {
   applicationName = "firecloud:leonardo"
   dataprocDockerImage = "gcr.io/broad-dsde-prod/leonardo-notebooks:prod"
+  defaultExecutionTimeout = "1800s"
```
Comment (rtitle) [inline src/main/resources/reference.conf]: Typesafe config can parse:
```
defaultExecutionTimeout = 30 minutes
```
We use this pattern elsewhere in the config file.

**Review by rtitle — CHANGES_REQUESTED** [review]
One change requested to represent it as a `FiniteDuration` in Leo code, and convert to a Google duration in `HttpGoogleDataprocDAO`.

**Comment by MatthewBemis** [comment]
@rtitle ready for re-review. see the note i left atop `finiteDurationToGoogleDuration`. if you think it matters that it won't support fractional seconds, i can add that support

**Review by rtitle — APPROVED** [review]
Looks great! cc @abaumann FYI this PR ups the cluster initialization timeout to 30 mins by default.

Can we do a quick test to verify that this works?

1. Create a bash script which sleeps for ~15 minutes and add it to a bucket
2. Create a cluster, specifying the gs:// path to that script
3. Make sure the cluster creation succeeds

I think a manual test is fine in this case, as opposed to an automation test (which would increase test execution time).

**Comment by rtitle** [comment]
PS - yeah I think second-granularity is fine.

**Pull request merged**